{
  if (done) {
    innerDone();
  }
  exception=throwable;
  setException(throwable);
}
